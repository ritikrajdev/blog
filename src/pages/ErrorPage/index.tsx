import React from 'react';
import { useParams } from 'react-router-dom';

export default function ErrorPage() {
  const { errorCode } = useParams();
  const _errorCode = isNaN(Number(errorCode)) ? undefined : errorCode;

  return (
    <div style={{ textAlign: 'center' }}>
      <p>Oh No ! Something went wrong</p>
      {_errorCode && (
        <p data-testid='error-code-container'>
          A Error Occoured with Error Code {_errorCode}
        </p>
      )}
    </div>
  );
}
