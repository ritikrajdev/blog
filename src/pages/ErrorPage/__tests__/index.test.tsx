import React from 'react';
import ErrorPage from '..';
import { render } from '@testing-library/react';
import { useParams } from 'react-router-dom';

jest.mock('react-router-dom');
const mockUseParams = useParams as jest.MockedFunction<typeof useParams>;

describe('ErrorPage', () => {
  it('should render correctly without param', () => {
    mockUseParams.mockReturnValue({});
    const screen = render(<ErrorPage />);
    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should render errorCode correctly with correct param', () => {
    mockUseParams.mockReturnValue({ errorCode: '123' });
    const screen = render(<ErrorPage />);

    const errorCodeContainer = screen.getByTestId('error-code-container');
    expect(errorCodeContainer).toBeTruthy();

    expect(errorCodeContainer.textContent).toContain('123');

    expect(screen.asFragment()).toMatchSnapshot();
  });

  it('should render errorCode correctly with incorrect param', () => {
    mockUseParams.mockReturnValue({ errorCode: '123x' });
    const screen = render(<ErrorPage />);

    const errorCodeContainer = screen.queryByTestId('error-code-container');
    expect(errorCodeContainer).toBeFalsy();

    expect(screen.asFragment()).toMatchSnapshot();
  });
});
