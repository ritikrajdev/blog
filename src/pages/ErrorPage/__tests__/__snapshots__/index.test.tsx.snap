// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render correctly without param 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <p>
      Oh No ! Something went wrong
    </p>
  </div>
</DocumentFragment>
`;

exports[`ErrorPage should render errorCode correctly with correct param 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <p>
      Oh No ! Something went wrong
    </p>
    <p
      data-testid="error-code-container"
    >
      A Error Occoured with Error Code 123
    </p>
  </div>
</DocumentFragment>
`;

exports[`ErrorPage should render errorCode correctly with incorrect param 1`] = `
<DocumentFragment>
  <div
    style="text-align: center;"
  >
    <p>
      Oh No ! Something went wrong
    </p>
  </div>
</DocumentFragment>
`;
